core/txpool/list_test.go

```diff
@@ -78,3 +81,68 @@ func BenchmarkListAdd(b *testing.B) {
 		}
 	}
 }
+
+func TestFilterTxConditional(t *testing.T) {
+	t.Parallel()
+
+	// Create an in memory state db to test against.
+	memDb := rawdb.NewMemoryDatabase()
+	db := state.NewDatabase(memDb)
+	state, _ := state.New(common.Hash{}, db, nil)
+
+	// Create a private key to sign transactions.
+	key, _ := crypto.GenerateKey()
+
+	// Create a list.
+	list := newList(true)
+
+	// Create a transaction with no defined tx options
+	// and add to the list.
+	tx := transaction(0, 1000, key)
+	list.Add(tx, DefaultConfig.PriceBump)
+
+	// There should be no drops at this point.
+	// No state has been modified.
+	drops := list.FilterTxConditional(state)
+
+	if count := len(drops); count != 0 {
+		t.Fatalf("got %d filtered by TxOptions when there should not be any", count)
+	}
+
+	// Create another transaction with a known account storage root tx option
+	// and add to the list.
+	tx2 := transaction(1, 1000, key)
+
+	var options types.OptionsAA4337
+
+	options.KnownAccounts = types.KnownAccounts{
+		common.Address{19: 1}: &types.Value{
+			Single: common.HexToRefHash("0xe734938daf39aae1fa4ee64dc3155d7c049f28b57a8ada8ad9e86832e0253bef"),
+		},
+	}
+
+	state.SetState(common.Address{19: 1}, common.Hash{}, common.Hash{30: 1})
+	tx2.PutOptions(&options)
+	list.Add(tx2, DefaultConfig.PriceBump)
+
+	// There should still be no drops as no state has been modified.
+	drops = list.FilterTxConditional(state)
+
+	if count := len(drops); count != 0 {
+		t.Fatalf("got %d filtered by TxOptions when there should not be any", count)
+	}
+
+	// Set state that conflicts with tx2's policy
+	state.SetState(common.Address{19: 1}, common.Hash{}, common.Hash{31: 1})
+
+	// tx2 should be the single transaction filtered out
+	drops = list.FilterTxConditional(state)
+
+	if count := len(drops); count != 1 {
+		t.Fatalf("got %d filtered by TxOptions when there should be a single one", count)
```

Comment:
Same comment to use require as above.

Reply:
[Done](https://github.com/maticnetwork/bor/pull/945/commits/bce561598fb19c756b861d73a2168e79e1409a6f).